packages/arb-token-bridge-ui/src/components/common/AddCustomChain.tsx

```diff
@@ -109,52 +183,69 @@ export const AddCustomChain = () => {
         throw new Error('JSON input is empty.')
       }
 
-      const customChain = mapOrbitConfigToOrbitChain(data)
+      // validate config
+      ZodOrbitConfig.parse(data)
 
-      if (!allowedParentChainIds.includes(Number(customChain.partnerChainID))) {
-        throw new Error(
-          `Invalid partnerChainID ${customChain.partnerChainID}. Only Arbitrum testnet parent chains are allowed.`
-        )
-      }
+      const customChain = mapOrbitConfigToOrbitChain(data)
+      // Orbit config has been validated and will be added to the custom list after page refreshes
+      // let's still try to add it here to handle eventual errors
+      addCustomChain({ customChain })
 
       saveCustomChainToLocalStorage(customChain)
       // reload to apply changes
       location.reload()
     } catch (error: any) {
       setError(error.message ?? 'Something went wrong.')
+      setAddingChain(false)
     }
   }
 
   return (
     <>
       <textarea
         onChange={e => setChainJson(e.target.value)}
-        placeholder="Insert your Orbit JSON Config here."
+        placeholder="Insert the JSON configuration from the `outputInfo.json` file that's generated at the end of the custom Orbit chain deployment."
         className="min-h-[100px] w-full rounded-lg px-4 py-2 text-sm font-light text-black"
       />
-      {error && <span className="text-sm text-error">{error}</span>}
+      {error && (
+        <div className="relative">
+          <pre className="scroll mb-2 max-h-[400px] overflow-scroll rounded-lg border border-white/20 bg-white/5 p-4 text-sm text-error">
```

Comment:
should use `overflow-auto` otherwise the scrollbars show all the time

Reply:
updated